src/nix/flake.cc

```diff
@@ -24,8 +24,11 @@ using namespace nix;
 using namespace nix::flake;
 using json = nlohmann::json;
 
+struct CmdFlakeUpdate;
 class FlakeCommand : virtual Args, public MixFlakeOptions
 {
+    friend struct CmdFlakeUpdate;
+
     std::string flakeUrl = ".";
```

Comment:
```suggestion
protected:

    std::string flakeUrl = ".";
```

I wouldn't want to bless `CmdFlakeUpdate` here, so let's just make this protected if we allow subcommands to touch it

Reply:
Yeah that's reasonable, fixed!